jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/internal/memory/GCHeapSummaryHandler.java

```diff
@@ -105,8 +108,9 @@ private void recordValues(RecordedEvent before, RecordedEvent after) {
       memoryHistogram.record(after.getLong(HEAP_USED), ATTR_MEMORY_USED);
     }
     if (after.hasField(HEAP_SPACE)) {
-      if (after.getValue(HEAP_SPACE) instanceof RecordedObject ro) {
-        memoryHistogram.record(ro.getLong(COMMITTED_SIZE), ATTR_MEMORY_COMMITTED);
+      if (after.getValue(HEAP_SPACE) instanceof RecordedObject heapSpace) {
+        memoryHistogram.record(heapSpace.getLong(COMMITTED_SIZE), ATTR_MEMORY_COMMITTED);
+        memoryHistogram.record(heapSpace.getLong(RESERVED_SIZE), ATTR_MEMORY_RESERVED);
```

Comment:
Is there a guarantee that `COMMITTED_SIZE` and `RESERVED_SIZE` are present on?

Reply:
Good catch. Fixed.